draft-kinnear-webtransport-http2.md

```diff
@@ -63,459 +78,248 @@ Discussion of this draft takes place on the WebTransport mailing list
+transmit a datagram. WT_DATAGRAM frames are sent with Stream Identifier 0.
 
-DATA + STREAM_ID = 5 + END_STREAM
-WebTransport Data
 ~~~
-
-An example of the server initiating a WebTransport Stream follows. The only
-difference here is the endpoint that sends the first WTHEADERS frame.
-
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++---------------+
+|Pad Length? (8)|
++-+-------------+-----------------------------------------------+
+|R|                     Session ID (31)                         |
++-+-------------------------------------------------------------+
+|                            Data (*)                         ...
++---------------------------------------------------------------+
+|                           Padding (*)                       ...
++---------------------------------------------------------------+
 ~~~
-[[ From Client ]]                   [[ From Server ]]
+{: #fig-wt_datagram title="WT_STREAM Frame Format"}
 
-SETTINGS
-SETTINGS_ENABLE_CONNECT_[..] = 1
-SETTINGS_ENABLE_WEBTRANSPORT = 1
+The WT_DATAGRAM frame define the following fields:
 
-                                    SETTINGS
-                                    SETTINGS_ENABLE_CONNECT_[..] = 1
-                                    SETTINGS_ENABLE_WEBTRANSPORT = 1
+   Pad Length:  An 8-bit field containing the length of the frame
+      padding in units of octets.  This field is conditional (as
+      signified by a "?" in the diagram) and is only present if the
+      PADDED flag is set.
 
-HEADERS + END_HEADERS
-+ STREAM_ID = 3
-:method = CONNECT
-:protocol = webtransport
-:scheme = https
-:path = /
-:authority = server.example.com
+   Session ID:  An unsigned 31-bit integer that identifies the
+      stream Connect Stream for this Web Transport stream.  The
+      Session ID MUST be MUST be an open stream negotiated via the
+      extended CONNECT protocol with a `:protocol` value of
+      "webtransport".
 
-                                    HEADERS + END_HEADERS
-                                    + STREAM_ID = 3
-                                    :status = 200
+   Data:  Application data.  The amount of data is the remainder of the
+      frame payload after subtracting the length of the other fields
+      that are present.
 
-                                    WTHEADERS + END_HEADERS
-                                    + STREAM_ID = 2
-                                    + CONNECT_STREAM = 3
-                                    :method = GET
-                                    :scheme = https
-                                    :path = /
-                                    :authority = client.example.com
+The WT_DATAGRAM frame defines the following flags:
 
-WTHEADERS + END_HEADERS
-+ STREAM_ID = 2
-+ CONNECT_STREAM = 3
-:status = 200
+   UNIDIRECTIONAL (0x1):  When set, the stream begins in the
+      "half-closed (remote)" state.
```

Comment:
Does this part stay here even for datagrams?

Reply:
oops